src/Orleans.GrainDirectory.Redis/RedisGrainDirectory.cs

```diff
@@ -43,7 +51,12 @@ public async Task<GrainAddress> Register(GrainAddress address)
 
         public async Task Unregister(GrainAddress address)
         {
-            await this.database.ScriptEvaluateAsync(this.deleteScript, new { key = address.GrainId, val = JsonConvert.SerializeObject(address) });
+            var key = GetKey(address.GrainId);
+
+            var tx = this.database.CreateTransaction();
+            tx.AddCondition(Condition.StringEqual(key, JsonConvert.SerializeObject(address)));
+            tx.KeyDeleteAsync(key).Ignore();
+            await tx.ExecuteAsync();
```

Comment:
I believe we need to check the return value and potentially the condition: https://github.com/OrleansContrib/Orleans.Redis/blob/15b63ca48f11b715c6d05da1a60e85c9f8812247/src/Orleans.Clustering.Redis/RedisMembershipTable.cs#L75

Reply:
Although, I suppose in this case we don't care if it actually was deleted

Reply:
Maybe we can log it if we want? The current implementation has no log at all, it might be useful in the future to have some